sdk/core/Azure.Core/src/Pipeline/BearerTokenChallengeAuthenticationPolicy.cs

```diff
@@ -16,11 +18,18 @@ namespace Azure.Core.Pipeline
     /// A policy that sends an <see cref="AccessToken"/> provided by a <see cref="TokenCredential"/> as an Authentication header.
     /// Note: This class is currently in preview and is therefore subject to possible future breaking changes.
     /// </summary>
-    internal class BearerTokenChallengeAuthenticationPolicy : HttpPipelinePolicy
+    public class BearerTokenChallengeAuthenticationPolicy : HttpPipelinePolicy
     {
+        /// <summary>
+        /// The scopes currently configured for token requests to the credential
+        /// </summary>
+        protected ReadOnlyCollection<string> Scopes => _readOnlyScopes ??= Array.AsReadOnly(_scopes);
+
+        private string[] _scopes;
+        private ReadOnlyCollection<string>? _readOnlyScopes;
+
         private readonly AccessTokenCache _accessTokenCache;
-        protected string[] Scopes { get; private set; }
-        private readonly ValueTask<bool> _falseValueTask = new ValueTask<bool>(Task.FromResult(false));
+        private readonly ValueTask<bool> _falseValueTask = new(Task.FromResult(false));
```

Comment:
Does caching the value task actually save anything here since it's a struct? Why not just do the following in `AuthorizeRequestOnChallengeAsync`?

``` c#
return new ValueTask(false);
```

Reply:
or just `return default;`